### PR TITLE
Enh/client config manager

### PR DESCRIPTION
**Pull request by bob0005**
## Release Notes

- **New Features**
	- Introduced `max_troop_count` property in troop configuration for larger troop counts.
	- Added new methods in the configuration manager for retrieving various game settings, including stamina costs and troop configurations.

- **Improvements**
	- Enhanced troop health management and battle status checks for better performance.
	- Streamlined configuration management across multiple components, replacing static references with dynamic calls to the configuration manager.

- **Bug Fixes**
	- Adjusted calculations for troop limits and stamina costs to ensure accurate gameplay experience.

- **Documentation**
	- Updated internal documentation to reflect changes in configuration management and component interactions.

- **Chores**
	- Removed deprecated imports and cleaned up code for better maintainability.

### CODE REVIEW

File: client/src/dojo/modelManager/ConfigManager.ts [hunk 1]
```diff
@@ -16,32 +31,164 @@ export class ClientConfigManager {
...
-  getStaminaTravelConfig() {
+  getTravelStaminaCost() {
+    if (!this.isDojoSet()) return 0;
+
     const staminaConfig = getComponentValue(
       this.components.TravelStaminaCostConfig,
       getEntityIdFromKeys([WORLD_CONFIG_ID, BigInt(TravelTypes.Travel)]),
     );
-    return staminaConfig?.cost!;
+    return staminaConfig?.cost ?? 0;
```
[the remark this answers is not in the document]
Comment (ponderingdemocritus) [reply client/src/dojo/modelManager/ConfigManager.ts:44]: @bob0005

**Comment by ponderingdemocritus** [comment]
this is looking good - need to test it all in the rc before merging this

**Comment by ponderingdemocritus** [comment]
Still TODO:

- RESOURCE_BUILDING_COSTS_SCALED + others
-- These should be looped in the ConfigManager into an object, then used around the app.

File: client/src/dojo/modelManager/utils/LeaderboardUtils.ts [hunk 2]
```diff
@@ -1,22 +1,7 @@
...
-      total +
-      (HYPERSTRUCTURE_RESOURCE_MULTIPLIERS[resource as keyof typeof HYPERSTRUCTURE_RESOURCE_MULTIPLIERS] ?? 0) * amount
-    );
-  },
-  0,
-);
+import { ClientConfigManager } from "../ConfigManager";
 
 function getResourceMultiplier(resourceType: ResourcesIds): number {
   return HYPERSTRUCTURE_RESOURCE_MULTIPLIERS[resourceType] ?? 0;
```
Comment (ponderingdemocritus) [inline client/src/dojo/modelManager/utils/LeaderboardUtils.ts:7]: this is in config object - let's replace it too

File: client/src/hooks/helpers/useQuests.tsx [hunk 3]
```diff
@@ -67,15 +71,22 @@ export const useQuests = () => {
   return { quests };
 };
 
-const useQuestDependencies = () => {
-  const {
-    setup,
-    account: { account },
-  } = useDojo();
-
+const useQuestDependencies = (setup: SetupResult, account: Account | AccountInterface) => {
```
Comment (ponderingdemocritus) [inline client/src/hooks/helpers/useQuests.tsx:74]: chagne the naming of this

getQuestDependencies

File: client/src/three/systems/SystemManager.ts [hunk 4]
```diff
@@ -281,28 +275,29 @@ export class SystemManager {
...
+          resource_type: resource,
+          amount: !foundProgress ? 0 : divideByPrecision(Number(foundProgress.amount)),
+          percentage: !foundProgress
+            ? 0
+            : Math.floor((divideByPrecision(Number(foundProgress.amount)) / resourceCost!) * 100),
+          costNeeded: resourceCost,
+        };
+        percentage +=
+          (progress.amount *
+            HYPERSTRUCTURE_RESOURCE_MULTIPLIERS[
```
Comment (ponderingdemocritus) [inline client/src/three/systems/SystemManager.ts:294]: same as above - need to pull from config

File: client/src/ui/components/bank/Swap.tsx [hunk 5]
```diff
@@ -8,7 +9,14 @@ import { ResourceBar } from "@/ui/components/bank/ResourceBar";
 import Button from "@/ui/elements/Button";
 import { ResourceIcon } from "@/ui/elements/ResourceIcon";
 import { divideByPrecision, multiplyByPrecision } from "@/ui/utils/utils";
-import { ContractAddress, EternumGlobalConfig, ID, ResourcesIds, resources } from "@bibliothecadao/eternum";
+import {
+  ContractAddress,
+  DONKEY_ENTITY_TYPE,
+  EternumGlobalConfig,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/bank/Swap.tsx:15]: delete EternumGlobalConfig,

File: client/src/ui/components/construction/SelectPreviewBuilding.tsx [hunk 6]
```diff
@@ -3,17 +3,11 @@ import clsx from "clsx";
 import useUIStore from "@/hooks/store/useUIStore";
 import { Tabs } from "@/ui/elements/tab";
 import {
-  BUILDING_CAPACITY,
-  BUILDING_POPULATION,
   BUILDING_RESOURCE_PRODUCED,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/construction/SelectPreviewBuilding.tsx:6]: const should not be here

File: client/src/ui/components/construction/SelectPreviewBuilding.tsx [hunk 7]
```diff
@@ -37,7 +32,12 @@ import { ResourceCost } from "@/ui/elements/ResourceCost";
...
 import { unpackResources } from "@/ui/utils/packedData";
 import { hasEnoughPopulationForBuilding } from "@/ui/utils/realms";
-import { getEntityIdFromKeys, isResourceProductionBuilding, ResourceIdToMiningType } from "@/ui/utils/utils";
+import {
+  divideByPrecision,
+  getEntityIdFromKeys,
+  isResourceProductionBuilding,
+  ResourceIdToMiningType,
+} from "@/ui/utils/utils";
 import { BUILDING_COSTS_SCALED } from "@bibliothecadao/eternum";
```
Comment (ponderingdemocritus) [inline client/src/ui/components/construction/SelectPreviewBuilding.tsx:41]: remove const

File: client/src/ui/components/construction/SelectPreviewBuilding.tsx [hunk 8]
```diff
@@ -532,16 +533,14 @@ export const BuildingInfo = ({
 
   const buildingCost = getBuildingCosts(entityId ?? 0, dojo, buildingId) || [];
 
-  const population = BUILDING_POPULATION[buildingId as keyof typeof BUILDING_POPULATION] || 0;
-  const capacity = BUILDING_CAPACITY[buildingId as keyof typeof BUILDING_CAPACITY] || 0;
+  const buildingPopCapacityConfig = configManager.getBuildingPopConfig(buildingId);
+  const population = buildingPopCapacityConfig.population;
+  const capacity = buildingPopCapacityConfig.capacity;
+
   const resourceProduced = BUILDING_RESOURCE_PRODUCED[buildingId as keyof typeof BUILDING_RESOURCE_PRODUCED];
```
Comment (ponderingdemocritus) [inline client/src/ui/components/construction/SelectPreviewBuilding.tsx:540]: const

File: client/src/ui/components/hints/Combat.tsx [hunk 9]
```diff
@@ -1,9 +1,12 @@
+import { configManager } from "@/dojo/setup";
 import { Headline } from "@/ui/elements/Headline";
 import { ResourceIcon } from "@/ui/elements/ResourceIcon";
-import { EternumGlobalConfig, ResourcesIds, TROOPS_STAMINAS } from "@bibliothecadao/eternum";
+import { ResourcesIds, TROOPS_STAMINAS } from "@bibliothecadao/eternum";
```
Comment (ponderingdemocritus) [inline client/src/ui/components/hints/Combat.tsx:4]: troop staminas needs replacing into config

File: client/src/ui/components/hints/TheMap.tsx [hunk 10]
```diff
@@ -1,8 +1,9 @@
 import { ReactComponent as Lightning } from "@/assets/icons/common/lightning.svg";
+import { configManager } from "@/dojo/setup";
 import { Headline } from "@/ui/elements/Headline";
 import { ResourceIcon } from "@/ui/elements/ResourceIcon";
 import { multiplyByPrecision } from "@/ui/utils/utils";
-import { EternumGlobalConfig, ResourcesIds, TROOPS_FOOD_CONSUMPTION } from "@bibliothecadao/eternum";
+import { ResourcesIds, TROOPS_FOOD_CONSUMPTION } from "@bibliothecadao/eternum";
```
Comment (ponderingdemocritus) [inline client/src/ui/components/hints/TheMap.tsx:6]: TROOPS_FOOD_CONSUMPTION

File: client/src/ui/components/hints/WorldStructures.tsx [hunk 11]
```diff
@@ -1,13 +1,12 @@
+import { configManager } from "@/dojo/setup";
 import { Headline } from "@/ui/elements/Headline";
 import { ResourceCost } from "@/ui/elements/ResourceCost";
 import { formatTime } from "@/ui/utils/utils";
 import {
   findResourceById,
   HYPERSTRUCTURE_POINTS_PER_CYCLE,
   HYPERSTRUCTURE_TIME_BETWEEN_SHARES_CHANGE_S,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/hints/WorldStructures.tsx:8]: HYPERSTRUCTURE_TIME_BETWEEN_SHARES_CHANGE_S
HYPERSTRUCTURE_POINTS_PER_CYCLE

remove

File: client/src/ui/components/hyperstructures/HyperstructurePanel.tsx [hunk 12]
```diff
@@ -11,10 +11,9 @@ import {
 import useUIStore from "@/hooks/store/useUIStore";
 import Button from "@/ui/elements/Button";
 import TextInput from "@/ui/elements/TextInput";
-import { currencyIntlFormat, getEntityIdFromKeys } from "@/ui/utils/utils";
+import { currencyIntlFormat, getEntityIdFromKeys, multiplyByPrecision } from "@/ui/utils/utils";
 import {
   ContractAddress,
-  EternumGlobalConfig,
   HYPERSTRUCTURE_POINTS_PER_CYCLE,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/hyperstructures/HyperstructurePanel.tsx:17]: HYPERSTRUCTURE_POINTS_PER_CYCLE,
  HYPERSTRUCTURE_TOTAL_COSTS_SCALED,
  
  Remove

File: client/src/ui/components/quest/questDetails.tsx [hunk 13]
```diff
@@ -7,7 +8,6 @@ import {
   BASE_POPULATION_CAPACITY,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/quest/questDetails.tsx:8]: BASE_POPULATION_CAPACITY,

File: client/src/ui/components/quest/questDetails.tsx [hunk 14]
```diff
@@ -7,7 +8,6 @@ import {
...
   BuildingType,
   CapacityConfigCategory,
-  EternumGlobalConfig,
   QuestType,
   ResourcesIds,
   TROOPS_FOOD_CONSUMPTION,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/quest/questDetails.tsx:13]: TROOPS_FOOD_CONSUMPTION,

File: client/src/ui/components/trading/MarketOrderPanel.tsx [hunk 15]
```diff
@@ -11,6 +12,7 @@ import { ResourceIcon } from "@/ui/elements/ResourceIcon";
 import { currencyFormat, divideByPrecision, getTotalResourceWeight, multiplyByPrecision } from "@/ui/utils/utils";
 import {
   CapacityConfigCategory,
+  DONKEY_ENTITY_TYPE,
   EternumGlobalConfig,
```
Comment (ponderingdemocritus) [inline client/src/ui/components/trading/MarketOrderPanel.tsx:16]: EternumGlobalConfig

**Review by ponderingdemocritus — COMMENTED** [review]
solid work - found some remaining configs that need removing before we can mereg

File: client/src/dojo/modelManager/utils/LeaderboardUtils.ts (hunk 2 above)
Comment (bob0005) [reply client/src/dojo/modelManager/utils/LeaderboardUtils.ts:7]: This is a workaround for client tests, otherwise they break atm

**Review by ponderingdemocritus — APPROVED** [review]

File: client/src/hooks/helpers/useQuests.tsx (hunk 3 above)
Comment (ponderingdemocritus) [reply client/src/hooks/helpers/useQuests.tsx:74]: it's more a function thou, not a hook

**Pull request merged**